### PR TITLE
fix: Token endpoint response, remove deprecation and code linting

### PR DESCRIPTION
**Pull request by peppelinux**
temporary patch before migrating to idpy-oidc

Closes https://github.com/UniversitaDellaCalabria/SATOSA-oidcop/issues/29

### CODE REVIEW

File: satosa_oidcop/idpy_oidcop.py [hunk 1]
```diff
@@ -350,10 +350,17 @@ def token_endpoint(self, context: ExtendedContext):
             return self.send_response(proc_req)
         elif isinstance(proc_req, TokenErrorResponse):
             return self.send_response(JsonResponse(proc_req.to_dict(), status="403"))
-
+        
+        # TODO: remove when migrate to idpy-oidc
+        # PATCH https://github.com/UniversitaDellaCalabria/SATOSA-oidcop/issues/29
+        if isinstance (proc_req["response_args"].get('scope', str), list):
+            proc_req["response_args"]['scope'] =  ' '.join(proc_req["response_args"]['scope'])
```
Comment (vladimir-mencl-eresearch) [inline satosa_oidcop/idpy_oidcop.py:357]: This is the crux of the change and it very much makes sense and aligns with the spec [1]: convert a list to space-separated string.

Just a bit of nit-picking: is there a reason for mixing the double and single quotes in a single expression?

[1] https://www.rfc-editor.org/rfc/rfc6749#section-3.3

File: tests/conftest.py [hunk 2]
```diff
@@ -42,14 +54,19 @@ def get_instance(cls):
...
+            self._process = subprocess.Popen(['mongod', '--bind_ip', 'localhost',
+                                              '--port', str(self._port),
+                                              '--dbpath', self._tmpdir,
+                                              '--nojournal',
+                                              '--noauth',
+                                              '--syncdelay', '0'],
+                                             stdout=open('/tmp/mongo-temp.log', 'wb'),
+                                             stderr=subprocess.STDOUT)
+        except FileNotFoundError as e:
+            logger.warning("Mongodb executable not found, trying to use docker compose mongodb")
```
Comment (vladimir-mencl-eresearch) [inline tests/conftest.py:68]: I am confused by this message - how is this related to docker-compose?

Comment (peppelinux) [reply tests/conftest.py:68]: If mongodb Is not installed It Will be used the One executed in the docker compose

I found a misalignment between the readme and the code, regarding the execution of the tests

We can improve, this Is Just a step forward or, better, just a fix

File: satosa_oidcop/idpy_oidcop.py (hunk 1 above)
Comment (peppelinux) [reply satosa_oidcop/idpy_oidcop.py:357]: I should run Blake on this!
Thank you, I'll align the quotes to be always twice

File: tests/conftest.py (hunk 2 above)
Comment (vladimir-mencl-eresearch) [reply tests/conftest.py:68]: Sorry, still at a loss, maybe due to the wording.

Does the message say that it failed to run mongodb _inside docker compose_ - or it failed to find mongodb in the default path and then it will try via docker compose?

Comment (peppelinux) [reply tests/conftest.py:68]: the second,

it falied the run of mongod locally, and it assumes that you've follow the instructions in the readme and have correctly executed the docker compose.

We can improve here, so consider me open for a proposal

File: satosa_oidcop/idpy_oidcop.py (hunk 1 above)
Comment (vladimir-mencl-eresearch) [reply satosa_oidcop/idpy_oidcop.py]: Thanks, just saw a lot of changes from the lint.

Just wondering: what line-width is the linting set to?

Some of the code looked more readable to me on a single line than split.

It might be just a matter of preference - but making sure we're not formatting to "historical terminal screens".

File: tests/conftest.py (hunk 2 above)
Comment (vladimir-mencl-eresearch) [reply tests/conftest.py]: Thanks for bearing with me.

I now understand the logic in that code.

It was: "always run mongodb as a sub-process".

It now is:
* try running mongodb as a sub-process.
* if the executable is not found, assume in runs under docker with the port 27017 mapped from localhost to the docker container - and connect there.

Thinking about wording that might convey the right message: how about:
`Mongodb executable not found to start test MongoDB instance, trying to connect to docker mongodb` ?

Cheers,
Vlad

**Review by melanger — APPROVED** [review]
commit message prefix should be probably `style:` instead of `chore:` for code linting, but it does not have any effect anyway

File: satosa_oidcop/idpy_oidcop.py (hunk 1 above)
Comment (peppelinux) [reply satosa_oidcop/idpy_oidcop.py]: linting here
https://github.com/UniversitaDellaCalabria/SATOSA-oidcop/blob/token_endpoint_res/linting.sh

Comment (vladimir-mencl-eresearch) [reply satosa_oidcop/idpy_oidcop.py]: Thanks - 120 chars is reasonable.
